discord/app_commands/transformers.py

```diff
@@ -551,11 +551,18 @@ class Range:
         During type checking time this is equivalent to :obj:`typing.Annotated` so type checkers understand
         the intent of the code.
 
-        Some example ranges:
+        Some example ranges limiting numeric input:
 
-        - ``Range[int, 10]`` means the minimum is 10 with no maximum.
-        - ``Range[int, None, 10]`` means the maximum is 10 with no minimum.
-        - ``Range[int, 1, 10]`` means the minimum is 1 and the maximum is 10.
+        - ``Range[int, 10]`` means the input type is `int`, and the minimum is 10 with no maximum.
+        - ``Range[int, None, 10]`` means the input type is `int`, and the maximum is 10 with no minimum.
+        - ``Range[int, 1, 10]`` means the input type is `int`, and the minimum is 1 and the maximum is 10.
+        - ``Range[float, 1.0, 5.0]`` means the input type is `float`, and the minimum is 1.0 and the maximum is 5.0.
```

Comment:
```suggestion
        - ``Range[int, 10]`` means the minimum is 10 with no maximum.
        - ``Range[int, None, 10]`` means the maximum is 10 with no minimum.
        - ``Range[int, 1, 10]`` means the minimum is 1 and the maximum is 10.
        - ``Range[float, 1.0, 5.0]`` means the minimum is 1.0 and the maximum is 5.0.
        - ``Range[str, 1, 10]`` means the minimum length is 1 and the maximum length is 10.
```

The input type is incredibly explicit, repeating it doesn't clarify anything. I've added the string type in here to be more explicit that it's possible, but the section below this should just be removed.